hazelcast/src/main/java/com/hazelcast/instance/Node.java

```diff
@@ -210,6 +214,24 @@ public Node(HazelcastInstanceImpl hazelcastInstance, Config config, NodeContext
         }
     }
 
+    private ClassLoader getConfigClassloader(Config config) {
+        DynamicClassloadingConfig dynamicClassloadingConfig = config.getDynamicClassloadingConfig();
+        ClassLoader classLoader;
+        if (dynamicClassloadingConfig.isEnabled()) {
+            ClassLoader parent = config.getClassLoader();
+            final ClassLoader theParent = parent == null ? Node.class.getClassLoader() : parent;
+            classLoader = doPrivileged(new PrivilegedAction<DynamicClassLoader>() {
```

Comment:
What is the exact reason of using `doPrivileged()`? We don't have access control checks anywhere in code. Why is it required here? You can also explain above the method.